src/main/java/com/puppycrawl/tools/checkstyle/checks/modifier/RedundantModifierCheck.java

```diff
@@ -209,7 +209,12 @@ private void checkInterfaceModifiers(DetailAST ast) {
      */
     private void checkEnumConstructorModifiers(DetailAST ast) {
         final DetailAST modifiers = ast.findFirstToken(TokenTypes.MODIFIERS);
-        final DetailAST modifier = modifiers.getFirstChild();
+        DetailAST modifier = modifiers.getFirstChild();
+
+        while (modifier != null && modifier.getType() == TokenTypes.ANNOTATION) {
+            modifier = modifier.getNextSibling();
+        }
+
```

Comment:
Logic is not easy to understand, why first modifier is not modifier :) , it might be better to make method `getFirstModifierAst()` in this class, one day it will be moved to common Util class.